[Max Autotune][B200] Add addmm config to avoid test OOM

jananisriram opened the pull request:
Summary: Add a new `addmm` config that is small enough to not cause an OOM (out of memory error), since the configs for `blackwell_persistent_mm_configs`, which `addmm` used, are too large.

Test Plan: `test_max_autotune.py`

Differential Revision: D83378477




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

coconutruben commented on torch/_inductor/template_heuristics/triton.py:
```diff
@@ -2094,7 +2098,8 @@ class CUDABlackwellAddmmPersistentTMATemplateConfigHeuristic(
         super().__init__()
-        self.mm_configs = self.blackwell_persistent_mm_configs
+        # NOTE: to ensure that we pass tests, addmm needs a small config
+        self.mm_configs = self.blackwell_persistent_mm_configs + self.blackwell_persistent_addmm_configs
```
should we still keep the old ones in here for addmm then? or just make a smaller set?

jananisriram replied:
we probably want to use the same configs for addmm as we use for mm, so it doesn't make as much sense to me to duplicate the configs across both `mm_configs` and `addmm_configs`! i think adding the singular config here makes sense to avoid duplication.

coconutruben reviewed (approved):
LGTM you can take/reject the question but this looks safe and smart!

jananisriram committed:
[Max Autotune][B200] Add addmm config to avoid test OOM (#164020)

Summary:

Add a new `addmm` config that is small enough to not cause an OOM (out of memory error), since the configs for `blackwell_persistent_mm_configs`, which `addmm` used, are too large.

Test Plan: `test_max_autotune.py`

Differential Revision: D83378477

jananisriram committed:
[Inductor][FP8] Add op_name for ScaledMM TMA template heuristic (#164019)

Summary:

For H100s and below, add `op_name="scaled_mm"` to the template heuristic for `CUDAScaledTMATemplateConfigHeuristic` such that `scaled_mm` persistent + TMA tests do not default to the "mm" heuristics.

Test Plan: `test_max_autotune.py`

Differential Revision: D83390775

jananisriram commented:
@pytorchbot merge
